File: 0160. Intersection of Two Linked Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> ListNode:
        
        a = headA
        b = headB
        
        while a != b :
            if not a :
                a = headB   #headA 跑完接headB跑
            else:
                a = a.next
                
        
            if not b :
                b = headA   #headB 跑完接headA跑
            else:
                b = b.next
                
        return a #return a或b其中一個 因為交叉點一樣
            

File: 0160. Intersection of Two Linked Lists.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode GetIntersectionNode(ListNode headA, ListNode headB) {
        
        ListNode a = headA;
        ListNode b = headB;
        
        while (a != b) {
             if (a == null){
                a = headB;  //headA 跑完接headB跑
            }else{
                a = a.next;
            }
        
            if (b == null){
                b = headA; 
            }else{
                b = b.next;
            }
            
        }
             
        return a; //return a或b其中一個 因為交叉點一樣
                   
    }
}